services/management/backup/backups_service.go

```diff
@@ -0,0 +1,153 @@
+}
+
+// NewBackupsService creates new backups API service.
+func NewBackupsService(db *reform.DB, jobsService jobsService) *BackupsService {
+	return &BackupsService{
+		db:          db,
+		jobsService: jobsService,
+	}
+}
+
+// StartBackup starts on-demand backup.
+func (s *BackupsService) StartBackup(ctx context.Context, req *backupv1beta1.StartBackupRequest) (*backupv1beta1.StartBackupResponse, error) {
+	var err error
+	var artifact *models.Artifact
+	var location *models.BackupLocation
+	var svc *models.Service
+	var job *models.JobResult
+	var dsn string
+
+	errTX := s.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		svc, err = models.FindServiceByID(tx.Querier, req.ServiceId)
+		if err != nil {
+			return err
+		}
+
+		location, err = models.FindBackupLocationByID(tx.Querier, req.LocationId)
+		if err != nil {
+			return err
+		}
+
+		artifact, err = models.CreateArtifact(tx.Querier, models.CreateArtifactParams{
+			Name:       req.Name,
+			Vendor:     string(svc.ServiceType),
+			LocationID: location.ID,
+			ServiceID:  svc.ServiceID,
+			DataModel:  models.PhysicalDataModel,
+			Status:     models.PendingBackupStatus,
+		})
+		if err != nil {
+			return err
+		}
+
+		job, dsn, err = s.prepareBackupJob(req.ServiceId, artifact.ID, models.MySQLBackupJob)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if errTX != nil {
+		return nil, errTX
+	}
+
+	locationConfig := models.BackupLocationConfig{
+		PMMServerConfig: location.PMMServerConfig,
+		PMMClientConfig: location.PMMClientConfig,
+		S3Config:        location.S3Config,
+	}
+
+	switch svc.ServiceType {
+
+	case models.MySQLServiceType:
+		err = s.jobsService.StartMySQLBackupJob(job.ID, job.PMMAgentID, 0, req.Name, dsn, locationConfig)
+	case models.PostgreSQLServiceType:
+		fallthrough
+	case models.MongoDBServiceType:
+		fallthrough
+	case models.ProxySQLServiceType:
+		fallthrough
+	case models.HAProxyServiceType:
+		fallthrough
+	case models.ExternalServiceType:
+		fallthrough
+	default:
+		return nil, status.Errorf(codes.Unimplemented, "unsupported service: %s", svc.ServiceType)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return &backupv1beta1.StartBackupResponse{
+		ArtifactId: artifact.ID,
+	}, nil
+}
+
+func (s *BackupsService) prepareBackupJob(serviceID, artifactID string, jobType models.JobType) (*models.JobResult, string, error) {
+	var res *models.JobResult
+	var dsn string
+	var pmmAgentID string
+	e := s.db.InTransaction(func(tx *reform.TX) error {
+		agents, err := models.FindPMMAgentsForService(tx.Querier, serviceID)
+		if err != nil {
+			return err
+		}
+
+		if pmmAgentID, err = models.FindPmmAgentIDToRunActionOrJob("", agents); err != nil {
+			return err
+		}
+
+		if dsn, _, err = models.FindDSNByServiceIDandPMMAgentID(tx.Querier, serviceID, pmmAgentID, ""); err != nil {
+			return err
+		}
+
+		res, err = models.CreateJobResult(tx.Querier, pmmAgentID, jobType, &models.JobResultData{
+			MySQLBackup: &models.MySQLBackupJobResult{
```

Comment:
I would suggest to create new model `backup` which would contain reference to artifact and job.

Reply:
It was discussed, for now we keep it as is.